### PR TITLE
adding BIO functions

### PR DESCRIPTION
**Pull request by lfcrypto**

### CODE REVIEW

File: src/include.am [hunk 1]
```diff
@@ -63,7 +63,22 @@ src_libwolfssl_la_SOURCES += \
...
+               wolfcrypt/src/bio_m_conn.c \
+               wolfcrypt/src/bio_m_accept.c \
+               wolfcrypt/src/bio_m_dgram.c \
+               wolfcrypt/src/bio_m_null.c \
+               wolfcrypt/src/bio_f_sock.c \
+               wolfcrypt/src/bio_f_buff.c \
+               wolfcrypt/src/bio_f_cipher.c \
+               wolfcrypt/src/bio_f_b64.c \
+               wolfcrypt/src/bio_f_dgst.c \
+               wolfcrypt/src/bio_f_ssl.c
```
Comment (dgarske) [inline src/include.am]: Can we move these new openssl specific bio files into a new folder either in "/wolfcrypt/src/openssl/" or "/src/openssl/" depending on where its appropriate. Seems like "src/openssl" is the right place to match patterning of .h files.

File: wolfcrypt/src/bio.c [hunk 2]
```diff
@@ -0,0 +1,823 @@
...
+#include <stdarg.h>
+#include <stdio.h>
+
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/logging.h>
+
+#ifdef NO_INLINE
+    #include <wolfssl/wolfcrypt/misc.h>
+#else
+    #include <wolfcrypt/src/misc.c>
```
Comment (dgarske) [inline wolfcrypt/src/bio.c]: You'll need to add "#define WOLFSSL_MISC_INCLUDED" above the misc.c include.

File: src/include.am (hunk 1 above)
Comment (dgarske) [inline src/include.am]: Ignore that. Sorry, I wasn't looking at all the changes together and I see you've combined all them together in wolfcrypt/src/bio.c. That looks good to me.

Comment (lfcrypto) [inline src/include.am]: In fact only wolfcrypt/src/bio.c exist. All previous file where put in it. I forgot to remove in include.am. So all can be remove except wolfcrypt/src/bio.c

File: wolfcrypt/src/bio.c (hunk 2 above)
Comment (lfcrypto) [inline wolfcrypt/src/bio.c]: yes effectively

File: examples/client/client.c [hunk 3]
```diff
@@ -1401,7 +1401,7 @@ THREAD_RETURN WOLFSSL_THREAD client_test(void* args)
                    " nonblocking yet");
         } else {
             if (wolfSSL_Rehandshake(ssl) != SSL_SUCCESS) {
-                int  err = wolfSSL_get_error(ssl, 0);
+                err = wolfSSL_get_error(ssl, 0);
                 char buffer[WOLFSSL_MAX_ERROR_SZ];
```
Comment (dgarske) [inline examples/client/client.c]: Good find on the "int err" here, but let's keep all variable declarations at top for compiler compatibility, so move the "char buffer[]" up a line. Thanks!

**Comment by toddouska** [comment]
Closed as PR #461 replaces this.

**Pull request closed without merging**